cms/src/test/java/uk/nhs/digital/common/OrderedSearchDateDerivedDataFunctionTest.java

```diff
@@ -0,0 +1,122 @@
+import org.apache.jackrabbit.value.LongValue;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.jcr.Value;
+
+@RunWith(DataProviderRunner.class)
+public class OrderedSearchDateDerivedDataFunctionTest {
+
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("dd-MM-yyyy");
+    private static final String ORDERED_SEARCH_DATE = "orderedSearchDate";
+
+    private OrderedSearchDateDerivedDataFunction function = new OrderedSearchDateDerivedDataFunction();
+
+    @Test
+    public void testUpcoming() throws Exception {
+        Map<String, Value[]> result = testFunctionCompute("20-03-2020", false);
+
+        long expectedValue = -parseDate("20-03-2020").getTimeInMillis();
+
+        assertThat("Result is as expected", result, hasEntry(ORDERED_SEARCH_DATE, new Value[]{new LongValue(expectedValue)}));
+    }
+
+    @Test
+    public void testPublished() throws Exception {
+        Map<String, Value[]> result = testFunctionCompute("10-08-2010", true);
+
+        long expectedValue = parseDate("10-08-2010").getTimeInMillis();
+
+        assertThat("Result is as expected", result, hasEntry(ORDERED_SEARCH_DATE, new Value[]{new LongValue(expectedValue)}));
+    }
+
+    @Test
+    public void testNoAccessibleFlag() throws Exception {
+        Map<String, Value[]> result = testFunctionCompute("18-11-2013", null);
+
+        long expectedValue = parseDate("18-11-2013").getTimeInMillis();
+
+        assertThat("Result is as expected", result, hasEntry(ORDERED_SEARCH_DATE, new Value[]{new LongValue(expectedValue)}));
+    }
+
+    @Test
+    public void testNoDate() throws Exception {
+        assertNull(testFunctionCompute(null, true).get(ORDERED_SEARCH_DATE));
+
+        assertNull(testFunctionCompute(null, false).get(ORDERED_SEARCH_DATE));
+
+        assertNull(testFunctionCompute(null, null).get(ORDERED_SEARCH_DATE));
+    }
+
+    @Test
+    public void testOrder() throws Exception {
+        Map<String, Long> inputToOutput = new HashMap();
+        addData(inputToOutput, "25-02-2020", null);
+        addData(inputToOutput, "09-03-2020", true);
+        addData(inputToOutput, "10-03-2020", false);
+        addData(inputToOutput, "11-03-2020", true);
+        addData(inputToOutput, "12-03-2020", false);
+        addData(inputToOutput, "15-03-2020", null);
+        addData(inputToOutput, "20-05-2023", true);
+        addData(inputToOutput, "21-06-2023", false);
+
+        List<String> sortedDates = inputToOutput.entrySet().stream()
+            .sorted(Map.Entry.comparingByValue())
+            .map(Map.Entry::getKey)
+            .collect(Collectors.toList());
+
+        // We sort from highest to lowest so need to reverse
+        Collections.reverse(sortedDates);
+
+        // The order should be published first in descending order followed by upcoming in ascending date order
```

Comment:
This comment could actually be turned into an assertion message, replacing generic 'Order is as expected' one, thus communicating the expectation both in the code and in the test failure output.